stories/knobs/text.js

```diff
@@ -0,0 +1,125 @@
+import React from 'react'
+import Text from '../../src/components/text';
+
+import { selectV2, boolean } from '@storybook/addon-knobs';
+
+import { textSizes, textTopMargins, textColors, textFonts, textAligns, textDisplays } from '../../src/styleConfig';
+
+const { defaultProps } = Text;
+
+const sizesOptions = Object.keys(textSizes).reduce((acc, next) => {
+    acc[next] = next;
+    return acc;
+}, {});
+
+const topMarginsOptions = Object.keys(textTopMargins).reduce((acc, next) => {
+    acc[next] = next;
+    return acc;
+}, {});
+
+const colorsOptions = Object.keys(textColors).reduce((acc, next) => {
+    acc[next] = next;
+    return acc;
+}, {});
+
+const getOptionsForProp = prop => {
+    switch (prop) {
+        case 'color':
+            return colorsOptions;
+  
+        case 'size':
+            return sizesOptions;
+  
+        case 'topMargin':
+            return topMarginsOptions;
+
+        case 'align':
+            return textAligns
+
+        case 'font':
+            return textFonts;
+
+        case 'display':
+            return textDisplays;
+
+        default:
+            return {};
+    }
+};
+
+const getKnobTypeForProp = (prop, value) => {
+    switch (prop) {
+        case 'ellipsis':
+            return {
+            knobFunction: boolean,
+            args: [value.label, value.defaultValue]
+            };
+    
+        case 'noWrap':
+            return {
+            knobFunction: boolean,
+            args: [value.label, value.defaultValue]
+            };
+    
+        case 'underline':
+            return {
+            knobFunction: boolean,
+            args: [value.label, value.defaultValue]
+            };
+  
+        case 'topMargin':
+            return {
+            knobFunction: selectV2,
+            args: [value.label, value.options, value.defaultValue]
+            };
+    
+        case 'color':
+            return {
+            knobFunction: selectV2,
+            args: [value.label, value.options, value.defaultValue]
+            };
+    
+        case 'size':
+            return {
+            knobFunction: selectV2,
+            args: [value.label, value.options, value.defaultValue]
+            };
+
+        case 'font':
+            return {
+                knobFunction: selectV2,
+                args: [value.label, value.options, value.defaultValue]
+            }
+  
+        case 'align':
+            return {
+            knobFunction: selectV2,
+            args: [value.label, value.options, value.defaultValue]
+            };
+
+        case 'display':
+        return {
+            knobFunction: selectV2,
+            args: [value.label, value.options, value.defaultValue]
+            };
```

Comment:
There seems to be a few formatting issues. Do you have prettier enabled on your code editor?